Makefile

```diff
@@ -12,50 +29,35 @@ DOC_FILES := \
 FIGURE_FILES := \
 	img/media-types.png
 
-OUTPUT		?= output/
-DOC_FILENAME	?= oci-image-spec
-
-EPOCH_TEST_COMMIT ?= v0.2.0
-
 default: help
 
 help:
 	@echo "Usage: make <target>"
 	@echo
-	@echo " * 'docs' - produce document in the $(OUTPUT) directory"
+	@echo " * 'docs' - produce document in the $(OUTPUT_DIRNAME) directory"
 	@echo " * 'fmt' - format the json with indentation"
 	@echo " * 'validate' - build the validation tool"
 
 fmt:
 	for i in *.json ; do jq --indent 2 -M . "$${i}" > xx && cat xx > "$${i}" && rm xx ; done
 
-docs: $(OUTPUT)/$(DOC_FILENAME).pdf $(OUTPUT)/$(DOC_FILENAME).html
-
-$(OUTPUT)/$(DOC_FILENAME).pdf: $(DOC_FILES) $(FIGURE_FILES)
-	@mkdir -p $(OUTPUT)/ && \
-	cp -ap img/ $(shell pwd)/$(OUTPUT)/&& \
-	$(DOCKER) run \
-	-it \
-	--rm \
-	-v $(shell pwd)/:/input/:ro \
-	-v $(shell pwd)/$(OUTPUT)/:/$(OUTPUT)/ \
-	-u $(shell id -u) \
-	--workdir /input \
-	vbatts/pandoc -f markdown_github -t latex -o /$(OUTPUT)/$(DOC_FILENAME).pdf $(patsubst %,/input/%,$(DOC_FILES)) && \
+docs: $(OUTPUT_DIRNAME)/$(DOC_FILENAME).pdf $(OUTPUT_DIRNAME)/$(DOC_FILENAME).html
+
+ifeq "$(strip $(PANDOC))" ''
+$(OUTPUT_DIRNAME)/$(DOC_FILENAME).pdf $(OUTPUT_DIRNAME)/$(DOC_FILENAME).html:
+	$(error cannot build $@ without either pandoc or docker)
+else
+$(OUTPUT_DIRNAME)/$(DOC_FILENAME).pdf: $(DOC_FILES) $(FIGURE_FILES)
+	@mkdir -p $(OUTPUT_DIRNAME)/ && \
+	$(PANDOC) -f markdown_github -t latex -o $(PANDOC_DST)$@ $(patsubst %,$(PANDOC_SRC)%,$(DOC_FILES))
```

Comment:
I'll double check this morning

On Wed, Aug 24, 2016, 07:42 Graham Whaley <notifications@github.com> wrote:

> In Makefile
> <https://github.com/opencontainers/image-spec/pull/131#discussion_r76041071>
> :
>
> > -	-it \
> > -	--rm \
> > -	-v $(shell pwd)/:/input/:ro \
> > -	-v $(shell pwd)/$(OUTPUT)/:/$(OUTPUT)/ \
> > -	-u $(shell id -u) \
> > -	--workdir /input \
> > -	vbatts/pandoc -f markdown_github -t latex -o /$(OUTPUT)/$(DOC_FILENAME).pdf $(patsubst %,/input/%,$(DOC_FILES)) && \
> > +docs: $(OUTPUT_DIRNAME)/$(DOC_FILENAME).pdf $(OUTPUT_DIRNAME)/$(DOC_FILENAME).html
> > +
> > +ifeq "$(strip $(PANDOC))" ''
> > +$(OUTPUT_DIRNAME)/$(DOC_FILENAME).pdf $(OUTPUT_DIRNAME)/$(DOC_FILENAME).html:
> > +	$(error cannot build $@ without either pandoc or docker)
> > +else
> > +$(OUTPUT_DIRNAME)/$(DOC_FILENAME).pdf: $(DOC_FILES) $(FIGURE_FILES)
> > +	@mkdir -p $(OUTPUT_DIRNAME)/ && \
> > +	$(PANDOC) -f markdown_github -t latex -o $(PANDOC_DST)$@ $(patsubst %,$(PANDOC_SRC)%,$(DOC_FILES))
>
> I am afk for the next week, will look when I get back. The cp is present
> for the html generation is it not? Is the cp also necessary for the pdf?,
> as the images will be incorporated in the pdf file will they not?, or am I
> missing something?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/opencontainers/image-spec/pull/131/files/0947df9fe21e157d2dac0ec9721d6510e558720a#r76041071>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEF6ZEEsm-qHCnFnd7O9-0U2sUFdMOIks5qjC47gaJpZM4I0Mfq>
> .
>